File: Go/sqrt/69_test.go
package sqrt

import (
	"fmt"
	"testing"

	"github.com/stretchr/testify/assert"
)

func Test_mySqrt(t *testing.T) {
	tests := []struct {
		x    int
		want int
	}{
		{x: 4, want: 2},
		{x: 8, want: 2},
	}

	for i, tt := range tests {
		t.Run(fmt.Sprintf("case %d", i+1), func(t *testing.T) {
			got := mySqrt(tt.x)
			assert.Equal(t, tt.want, got)
		})
	}
}

File: TypeScript/sqrt/69.test.ts
import mySqrt from "./69";

interface tt {
  x: number;
  want: number;
}

const tests: tt[] = [
  { x: 4, want: 2 },
  { x: 8, want: 2 },
];

describe("sqrt", () => {
  tests.forEach((t, i) => {
    test(`case ${i + 1}`, () => {
      const result = mySqrt(t.x);
      expect(result).toBe(t.want);
    });
  });
});
